frontend/src/http/user.js

```diff
@@ -1,3 +1,22 @@
+//register a new user
+async function registerUser(username, email, password) {
+    const data = { name: username, email: email, password: password };
+    try {
+        const res = await fetch(
+            '/api/users',
+            {
+                method: 'POST',
+                body: JSON.stringify(data),
+            })
+        const jsonResponse = await res.json();
+        if (!res.ok) {
+            return jsonResponse.error;
```

Comment:
This should be `return jsonResponse`. Do a console.log on what jsonResponse is when there's an error. You'll see it's an object like `{"error": "..."}`. The handling of the error here is different than what you're doing on line 17. 

When this function has an error it will return either a plain string or an object with an error property. That doesn't seem easy to handle from the caller's perspective, they now need to embed in their logic: **if i have an error property, or the response is a plain string then we have an error, otherwise the response is good.**

Reply:
changed return